app/src/main/java/com/example/reduxtestapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.reduxtestapp.ui.home
+
+import com.example.reduxtestapp.redux.Action
+import com.example.reduxtestapp.redux.AppState
+import com.example.reduxtestapp.ui.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import org.reduxkotlin.Store
+
+class HomeViewModel(
+    store: Store<AppState>,
+): ViewModel(store) {
+
+    private val _uiState = MutableStateFlow(HomeViewState())
+    val uiState = _uiState.asStateFlow()
+
+    fun startPolling() {
+        dispatch(Action.Home.PollPrice)
+    }
+
+    fun stopPolling() {
+        dispatch(Action.Home.StopPollingPrice)
+    }
```

Comment:
this is leaning more towards mvvm now, which is not bad in itself, but let's stick to redux for this project so dispatch these actions from the screen instead of using these methods